staging/src/k8s.io/cloud-provider/app/controllermanager.go

```diff
@@ -182,8 +188,10 @@ func Run(c *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface
 	}
 
 	if !c.ComponentConfig.Generic.LeaderElection.LeaderElect {
-		run(context.TODO(), controllerInitializers)
-		panic("unreachable")
+		ctx, _ := wait.ContextForChannel(stopCh)
+		run(ctx, controllerInitializers)
```

Comment:
hehe 

https://github.com/kubernetes/kubernetes/blob/c09bf9623ec90b03e35e48af8b48bcb30f38bec9/staging/src/k8s.io/cloud-provider/app/controllermanager.go#L316

Reply:
the `run` seems that will be running forever, `startControllers` doesn't leverage the context neither the stopCh ... that is the other funny part, we duplicate stop signals :upside_down_face:

Reply:
They actually do utilize it:
- we're passing ctx.Done() to startControllers
- they are using it (as stopCh) in two places

But thanks for catching the select{} - fixed that now.